lib/railway_ui_web/views/message_view_helper.ex

```diff
@@ -1,5 +1,15 @@
 defmodule RailwayUiWeb.MessageViewHelper do
-  def decode_message_body(message_body) do
-    inspect(Jason.decode!(message_body))
+  def decode_message_body(%{message_type: message_type, encoded_message: encoded_message}) do
+    %{"encoded_message" => message_to_decode} = Jason.decode!(encoded_message)
+
+    try do
+      stuff = message_type
+      |> RailwayIpc.Core.Payload.module_from_type()
+      |> RailwayIpc.Core.Payload.decode_message(message_to_decode)
+      |> inspect(pretty: true)
+    rescue
+      UndefinedFunctionError -> "Unable to decode message due to missing message module"
```

Comment:
If a message type no longer exists in the app this will be raised by calling ` RailwayIpc.Core.Payload.decode_message(module, message_to_decode)`. In the test example I was using it was from calling the soon to be deleted AdmissionCreated message module.

Reply:
I'm happy to chat IRL on this, but seeing the encoded message is not something I think we will find useful. If we are unable to decode the encoded values what do we gain here? or maybe we render the message + the encoded message?

What do you mean this will fail out on every prod app? or do you just mean there will be some messages that fail and we should handle this at the **RailwayIpc.Core.Payload#decode_message/2** level upon the message module existing or not.

Reply:
@aturkewi Just made an update to use module check instead of error flow rendering.

Reply:
It looked like you were raising and rescuing if you couldn't decode. We were then passing an error object to the front end to display and I think that would have blown up when trying to convert it to a string?
And I was just saying I think the encryped message was a better fallback that way we at least have _some_ info and w would all know that if we saw an encrypted msg it is because it could not be decrypted so an explicit string saying it couldn't be decrypted didn't seem needed to me.
But as I mentioned in my approve comment, I think what you came up with will work well and I am very excited to have this in the code base :)